checkmedia.c

```diff
@@ -1,598 +1,205 @@
-      iso.digest.type = digest_sha1;
-      iso.digest.size = SHA1_DIGEST_SIZE;
-      iso.digest.name = "sha1";
-    }
-    else if(!strcasecmp(key, "sha224sum")) {
-      iso.digest.type = digest_sha224;
-      iso.digest.size = SHA224_DIGEST_SIZE;
-      iso.digest.name = "sha224";
-    }
-    else if(!strcasecmp(key, "sha256sum")) {
-      iso.digest.type = digest_sha256;
-      iso.digest.size = SHA256_DIGEST_SIZE;
-      iso.digest.name = "sha256";
-    }
-    else if(!strcasecmp(key, "sha384sum")) {
-      iso.digest.type = digest_sha384;
-      iso.digest.size = SHA384_DIGEST_SIZE;
-      iso.digest.name = "sha384";
-    }
-    else if(!strcasecmp(key, "sha512sum")) {
-      iso.digest.type = digest_sha512;
-      iso.digest.size = SHA512_DIGEST_SIZE;
-      iso.digest.name = "sha512";
-    }
-
-    if(iso.digest.type && value) {
-      if(strlen(value) >= iso.digest.size) {
-        for(u = 0 ; u < iso.digest.size; u++, value += 2) {
-           if(sscanf(value, "%2x", &u1) == 1) {
-             iso.digest.old[u] = u1;
-           }
-           else {
-             break;
-           }
-        }
-        if(u == iso.digest.size) iso.digest.got_old = 1;
-      }
-    }
-
-    if(value && !strcasecmp(key, "pad")) {
-      if(isdigit(*value)) iso.pad = strtoul(value, NULL, 0) << 1;
-    }
-
-    if(value && !strcasecmp(key, "skipsectors")) {
-      if(isdigit(*value)) {
-        iso.pad = strtoul(value, NULL, 0) << 1;
-        iso.pad_is_skip = 1;
-      }
+  if(opt.verbose) {
+    if(mediacheck_digest_valid(media->digest.iso)) {
+      printf(" iso %6s: %s\n", mediacheck_digest_name(media->digest.iso), mediacheck_digest_hex(media->digest.iso));
     }
-  }
-}
-
 
-/*
- * Remove leading & trailing spaces.
- * Modifies str!
- */
-char *no_extra_spaces(char *str)
-{
-  int i;
-
-  if(str) {
-    while(isspace(*str)) str++;
-    if((i = strlen(str))) {
-      while(i && isspace(str[i-1])) str[--i] = 0;
+    if(mediacheck_digest_valid(media->digest.part)) {
+      printf("part %6s: %s\n", mediacheck_digest_name(media->digest.part), mediacheck_digest_hex(media->digest.part));
     }
   }
 
-  return str;
-}
-
-
-void update_progress(unsigned size)
-{
-  static int last_percent = 0;
-  int percent;
-
-  if(!size) last_percent = 0;
-
-  percent = (size * 100) / (iso.size ?: 1);
-  if(percent != last_percent) {
-    last_percent = percent;
-    printf("\x08\x08\x08\x08%3d%%", percent); fflush(stdout);
+  if(mediacheck_digest_valid(media->digest.full)) {
+    printf("%11s: %s\n", mediacheck_digest_name(media->digest.full), mediacheck_digest_hex(media->digest.full));
   }
-}
 
+  i = mediacheck_digest_ok(media->digest.iso) || mediacheck_digest_ok(media->digest.part) ? 0 : 1;
 
-/*
- * Read 32 bit value at pointer address, little-endian.
- */
-uint32_t read_le32(const void *ptr)
-{
-  const uint8_t *s = ptr;
+  mediacheck_done(media);
 
-  return s[0] + (s[1] << 8) + (s[2] << 16) + (s[3] << 24);
+  return i;
```

Comment:
here I would call it result. Reusing variables is not so nice if it cause confusion about usage. I think gcc can already optimize it out and reuse registers when needed, so variable reusage is not needed.